### PR TITLE
Chore/containerize express

### PR DESCRIPTION
**Pull request by jwsmith24**
Set up Docker to build and run the whole system!

I updated the start script to handle all the setup, but check out the README updates for more info on what it's actually doing.

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -5,11 +5,12 @@
   "main": "src/app.ts",
   "type": "module",
   "scripts": {
-    "dev": "tsc && node --watch dist/app.js",
```
Comment (cmccand1) [inline backend/package.json:8]: Does this remove hot reloading from the config? Live debugging has been very nice so far

Comment (jwsmith24) [reply backend/package.json:8]: Was it working for you on the backend?

Comment (jwsmith24) [reply backend/package.json:8]: Technically tsc is just running once so even if you changed the typescript, dist/app/js wouldn't update yet.

I tried playing around with a couple other options for typescript but I didn't have any luck

Comment (jwsmith24) [reply backend/package.json:8]: Except actually I think I just thought of exactly what was wrong typing that standby

Comment (jwsmith24) [reply backend/package.json:8]: Alright yeah that did it. HMR is actually working properly on the backend and will recompile and restart the server anytime you save changes!

Comment (cmccand1) [reply backend/package.json:8]: > Was it working for you on the backend?

Errr you're right. It wasn't working for changes on the backend (took me way to long to realize that!) but for frontend it is very nice.

> Alright yeah that did it. HMR is actually working properly on the backend and will recompile and restart the server anytime you save changes!

Sweet! Thanks for adding that!

**Pull request merged**